### ISSUE TITLE
Why is DNX abusing my $env:home?

### ISSUE THREAD

**Issue by bitcrazed** [aspnet/Home#1253]
**TL;DR: I believe the .dnx folder should be moved from my user profile to ... elsewhere**

It drives me nuts when applications infect my Documents folder with their settings etc:
![image](https://cloud.githubusercontent.com/assets/961950/12490545/fcbcc07c-c02a-11e5-8d7b-fe099687bae9.png)

**Seriously!! IISExpress ... WHY ARE YOU IN MY DOCUMENTS FOLDER?**

Forgive me ... I digress ...

The flagrant abuse of my documents folder is bad enough, but now systems like DNX/Nuget/DotNet are abusing my `$env:UserProfile` folder:

Currently, DNX stores all its runtimes, DNVM/packages, packages and settings in `$env:Home/.dnx`. 

Most of these files are read-only packages consumed by my projects; packages downloaded from NuGet etc., which can be shared between projects and have no place in my personal folders. Similarly, the runtimes can be shared between projects and have no place in my personal folders.

> Note: Currently, I am only working with DNX RC1 Update 1, and my `.dnx` folder is already 1.3GB on disk! This volume of data may will greatly impact user home folder backup and enterprise home folder replication via DFS, etc.

![image](https://cloud.githubusercontent.com/assets/961950/12491270/176eaecc-c02f-11e5-9d15-b36d5eb5bc0d.png)

Today, @davidfowl [announced](https://github.com/aspnet/Announcements/issues/142) that `$env:home/.dnx/packages` are now going to be restored to $env:home/.nuget/packages.

So now my user profile folder is going to be littered with even more **application data**. GREAT! Not! 

PLEASE stop this madness and **move the .dnx folders to appropriate special folders that were created for this exact scenario**:

* Roamable user-specific settings etc. should be stored in $env:AppData
* Local user-specific settings etc. should be stored in $env:LocalAppData
* Shared app data should be stored in $env:ProgramData

What am I missing here?

**Comment by sandersaares** [comment]
I seem to recall it being a standard approach to litter the user's home directory with such files on Linux systems. Perhaps the idea here is to keep parity between Windows and Linux by using some general purpose approach that is applicable everywhere? I agree it is rather nasty, though.

**Comment by LokiMidgard** [comment]
This works great on my Linux system. I still can use my home folder because Linux hides all files that start with a dot. But on Windows my Profile became unusable :(
More than have half of the elements in my Profile are configurations and other Programm stuff.

I love this cross-platform stuff, but is there now way to have a cross-platform program that doesn't litter your home folder on Windows. Like hiding them by default. After all that's what Linux does. 
Or having an environment variable that points on Linux to Home and to AppData on Windows.

**Comment by bitcrazed** [comment]
Closing since the new CoreCLR & dotnet no longer litter my home folder with binaries :)
